File: 2015/02/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	paper, ribbon := 0, 0
	for scanner.Scan() {
		dims := make([]int, 3)
		fmt.Sscanf(scanner.Text(), "%dx%dx%d", &dims[0], &dims[1], &dims[2])
		sort.Ints(dims)
		w, l, h := dims[0], dims[1], dims[2]
		paper += 2*(w*l+l*h+w*h) + w*l
		ribbon += 2*(w+l) + w*l*h
	}
	fmt.Println("Part One:", paper)
	fmt.Println("Part Two:", ribbon)
}

File: 2015/02/solution.py
import sys

paper, ribbon = 0, 0
for line in sys.stdin:
    w, l, h = sorted(int(n) for n in line.split("x"))
    paper += 2 * (w * l + l * h + w * h) + w * l
    ribbon += 2 * (w + l) + w * l * h
sys.stdout.write(f"Part One: {paper}\n")
sys.stdout.write(f"Part Two: {ribbon}\n")
